resources/assets/coffee/react/_components/beatmapset-panel.coffee

```diff
@@ -50,16 +50,25 @@ class @BeatmapsetPanel extends React.PureComponent
     # arbitrary number
     maxDisplayedDifficulty = 10
 
+    condenseDifficulties = beatmapset.beatmaps.length > maxDisplayedDifficulty
+
     difficulties =
-      for own _mode, beatmaps of BeatmapHelper.group beatmapset.beatmaps[..maxDisplayedDifficulty - 1]
-        for b in beatmaps
-          div
-            className: 'beatmapset-panel__difficulty-icon'
-            key: b.id
-            el BeatmapIcon, beatmap: b
-
-    if beatmapset.beatmaps.length > maxDisplayedDifficulty
-      difficulties.push span key: 'over', "+#{(beatmapset.beatmaps.length - maxDisplayedDifficulty)}"
+      for own mode, beatmaps of BeatmapHelper.group beatmapset.beatmaps
+        if condenseDifficulties
+          [..., hardestDiff] = beatmaps
+          span
+            key: mode
+            style: 
```

Comment:
Lol, I don't know which is worse, but yeah, with the array looked cleaner :neutral_face: